pkg/costmodel/costmodel.go

```diff
@@ -193,53 +269,26 @@ func (cm *CostModel) ComputeCostData(start, end time.Time) (map[string]*CostData
 
 		// ErrorCollection is an collection of errors wrapped in a single error implementation
 		// We opt to not return an error for the sake of running as a pure exporter.
-		log.Warnf("ComputeCostData: continuing despite prometheus errors: %s", grp.Error())
-	}
-
-	defer measureTime(time.Now(), profileThreshold, "ComputeCostData: Processing Query Data")
-
-	nodes, err := cm.GetNodeCost()
-	if err != nil {
-		log.Warnf("GetNodeCost: no node cost model available: %s", err)
-		return nil, err
-	}
-
-	// Unmounted PVs represent the PVs that are not mounted or tied to a volume on a container
-	unmountedPVs := make(map[string][]*PersistentVolumeClaimData)
-	pvClaimMapping, err := GetPVInfoLocal(cm.Cache, clusterID)
-	if err != nil {
-		log.Warnf("GetPVInfo: unable to get PV data: %s", err.Error())
-	}
-	if pvClaimMapping != nil {
-		err = cm.addPVData(pvClaimMapping)
-		if err != nil {
-			return nil, err
-		}
-		// copy claim mappings into zombies, then remove as they're discovered
-		for k, v := range pvClaimMapping {
-			unmountedPVs[k] = []*PersistentVolumeClaimData{v}
-		}
+		return resRAMUsage, resCPUUsage, resNetZoneRequests, resNetRegionRequests, resNetInternetRequests, grp.Error()
 	}
 
-	networkUsageMap, err := GetNetworkUsageData(resNetZoneRequests, resNetRegionRequests, resNetInternetRequests, clusterID)
-	if err != nil {
-		log.Warnf("Unable to get Network Cost Data: %s", err.Error())
-		networkUsageMap = make(map[string]*NetworkUsageData)
-	}
+	return resRAMUsage, resCPUUsage, resNetZoneRequests, resNetRegionRequests, resNetInternetRequests, nil
+}
 
+func getContainerCostData(podlist []*clustercache.Pod, resRAMUsage, resCPUUsage []*source.ContainerMetricResult, pvClaimMapping map[string]*PersistentVolumeClaimData, unmountedPVs map[string][]*PersistentVolumeClaimData, networkUsageMap map[string]*NetworkUsageData, podDeploymentsMapping, podServicesMapping map[string]map[string][]string, namespaceLabelsMapping, namespaceAnnotationsMapping map[string]map[string]string, nodes map[string]*costAnalyzerCloud.Node, clusterMap clusters.ClusterMap, cp costAnalyzerCloud.Provider, clusterID string) (map[string]*CostData, map[string]*costAnalyzerCloud.Node, map[string]*CostData, error) {
```

Comment:
Feel the same way about this function - I do not think we've gained anything from adding a function which accepts 10+ parameters - we're just moving variables around.